src/plots/cartesian/axes.js

```diff
@@ -4211,3 +4226,27 @@ function hideCounterAxisInsideTickLabels(ax, opts) {
         }
     }
 }
+
+function incrementShift(ax, shiftVal, axShifts) {
+    var shiftValAdj = ax.side === 'right' ? shiftVal : -shiftVal;
+    if(!(ax.overlaying in axShifts)) {
+        axShifts[ax.overlaying] = {};
+    }
+    if(!(ax.side in axShifts[ax.overlaying])) {
+        axShifts[ax.overlaying][ax.side] = 0;
+    }
+    axShifts[ax.overlaying][ax.side] += shiftValAdj;
+    return axShifts;
+}
+
+function setShiftVal(ax, axShifts) {
+    var shiftVal;
+    if(typeof(ax.shift) === 'number') {
+        shiftVal = ax.shift;
+    } else if(ax.shift === true) {
+        shiftVal = axShifts[ax.overlaying][ax.side];
+    } else {
+        shiftVal = 0;
+    }
+    return shiftVal;
```

Comment:
You may consider rewriting this function like this: 
```js
function setShiftVal(ax, axShifts) {
    return ax.shift === true ? 
        axShifts[ax.overlaying][ax.side] : 
        (ax.shift || 0);
}
```

Reply:
Yes, much better!